libexec/rtld-elf/aarch64/rtld_start.S

```diff
@@ -185,7 +166,16 @@ ENTRY(C18N_SYM(_rtld_bind_start))
 	ldr	PTR(0), [PTR(16), #-PTR_WIDTH]
 
 	/* Call into rtld */
+#ifdef CHERI_LIB_C18N
+	ldr	c10, c18n_rtld_bind
```

Comment:
Also, c10 isn't saved and restored, so on return the caller will have access to this function pointer. I assume that's not desired. (In the tlsdesc case c10 already is saved and restored, because almost every register is callee-saved for that special resolver calling convention)

Reply:
The trampoline of `c18n_rtld_bind` will take care of trashing c10, so it should be safe.

Reply:
But then proceed to restore it on the return path back to exactly the value loaded from c18n_rtld_bind.

Reply:
Sorry I'm confused. Do you mean the trampoline actually restores the value of c10 on its return path? But I don't think it does that.

Reply:
I understand now. Can you please add a comment that c10 is a call-clobbered register so the trampoline is guaranteed to clear it?